src/sdk/bucketspace_fs.js

```diff
@@ -388,6 +388,13 @@ class BucketSpaceFS extends BucketSpaceSimpleFS {
                         name,
                         full_path: path.join(this.fs_root, namespace_bucket_config.write_resource.path) // includes write_resource.path + bucket name (s3 flow)
                     }, object_sdk);
+                } else if (namespace_bucket_config){
+                    // TODO check only for single instance. 
```

Comment:
what this TODO means?

Reply:
@romayalon bucket delete should not happen even if there is a single file in bucket, in that case we dont need to iterate through complete files, It was a suggestion from Liran. Will update the comment with more explaining words.

Reply:
that reminds me of something that we need to handle here, if the directory only contains the internal hidden directory we should allow the deletion
few of our internal dirs are - 
get_bucket_tmpdir(), versions directory

Reply:
I asked for the TODO, when we will implement the object list optimization we can call readdir whiteout iterating over all the files. 
Then we should implement it also here, as this readdir is a performance hazard... 
If we have a lot of files/objects, then failing this delete bucket for the reason that the bucket is not empty will take a lot of time.

Reply:
currently on Namespace FS we do like follows - 
```
  const list = await this.list_objects({ ...params, limit: 1 }, object_sdk);

  if (list && list.objects && list.objects.length > 0) {
      throw new RpcError('NOT_EMPTY', 'underlying directory has files in it');
  }
```
I guess that what @liranmauda means, list_objects ignores also the hidden directories @naveenpaul1 you can do this here as well, just replace this with the correct namespace

Reply:
@romayalon This is a great sulotion

Reply:
@romayalon @liranmauda updated code with `ns.list_object()` to get the dir items